index/server/pkg/server/endpoint.go

```diff
@@ -80,169 +85,132 @@ func serveHealthCheck(c *gin.Context) {
 func serveDevfileWithVersion(c *gin.Context) {
 	name := c.Param("name")
 	version := c.Param("version")
-
-	var index []indexSchema.Schema
-	bytes, err := ioutil.ReadFile(indexPath)
-	if err != nil {
-		log.Print(err.Error())
-		c.JSON(http.StatusInternalServerError, gin.H{
-			"error":  err.Error(),
-			"status": fmt.Sprintf("failed to pull the devfile of %s", name),
-		})
-		return
-	}
-	err = json.Unmarshal(bytes, &index)
-	if err != nil {
-		log.Print(err.Error())
-		c.JSON(http.StatusInternalServerError, gin.H{
-			"error":  err.Error(),
-			"status": fmt.Sprintf("failed to pull the devfile of %s", name),
-		})
-		return
-	}
-
-	// minSchemaVersion and maxSchemaVersion will only be applied if looking for latest stack version
-	if version == "latest" {
-		minSchemaVersion := c.Query("minSchemaVersion")
-		maxSchemaVersion := c.Query("maxSchemaVersion")
-		// check if schema version filters are in valid format.
-		// should only include major and minor version. e.g. 2.1
-		if minSchemaVersion != "" {
-			matched, err := regexp.MatchString(`^([2-9])\.([0-9]+)$`, minSchemaVersion)
-			if !matched || err != nil {
-				c.JSON(http.StatusInternalServerError, gin.H{
-					"status": fmt.Sprintf("minSchemaVersion %s is not valid, should only include major and minor version. %v", minSchemaVersion, err),
-				})
-				return
+	bytes, devfileIndex := fetchDevfile(c, name, version)
+
+	if len(bytes) != 0 {
+		// Track event for telemetry.  Ignore events from the registry-viewer and DevConsole since those are tracked on the client side
+		if enableTelemetry && !util.IsWebClient(c) {
+
+			user := util.GetUser(c)
+			client := util.GetClient(c)
+
+			err := util.TrackEvent(analytics.Track{
+				Event:   eventTrackMap["view"],
+				UserId:  user,
+				Context: util.SetContext(c),
+				Properties: analytics.NewProperties().
+					Set("name", name).
+					Set("type", string(devfileIndex.Type)).
+					Set("registry", registry).
+					Set("client", client),
+			})
+			if err != nil {
+				log.Println(err)
 			}
 		}
-		if maxSchemaVersion != "" {
-			matched, err := regexp.MatchString(`^([2-9])\.([0-9]+)$`, maxSchemaVersion)
-			if !matched || err != nil {
-				c.JSON(http.StatusInternalServerError, gin.H{
-					"status": fmt.Sprintf("maxSchemaVersion %s is not valid, should only include major and minor version. %v", maxSchemaVersion, err),
-				})
-				return
-			}
+		c.Data(http.StatusOK, http.DetectContentType(bytes), bytes)
```

Comment:
The 200 status code is returned if `len(bytes) != 0`, since `fetchDevfile` takes care of the error responses within and does not return the errors the source does not do anything if unsuccessful. If unsuccessful, `len(bytes) == 0` such as in the case with [endpoint.go#L374](https://github.com/michael-valdron/registry-support/blob/bcf0c8789bbf43fdebce15cc2db87c7f4ce16f2d/index/server/pkg/server/endpoint.go#L374).